{{cookiecutter.package_name}}/src/{{cookiecutter.package_name}}/tools/cli.py

```diff
@@ -0,0 +1,125 @@
+    return True if len(loc_proper) else False, Location(loc_proper, loc_sanitized)
+
+
+def get_sanitized_locations(single_loc : str, loc_file : Path) -> List[Location]:
+    """ Naming convention is to lowercase the location string, replace spaces with underscores,
+        and capitalize the first letter. No manipulation happens in the template code.
+    """
+    def helper(item):
+        valid, loc = sanitize(item)
+        if valid:
+            locs.append(loc)
+
+    locs = []
+    if len(single_loc):
+        helper(single_loc)
+    else:
+        with loc_file.open('r') as infile:
+            for line in infile:
+                helper(line)
+    return locs
+    
+    
+def args_pass(template: Path, locations_file: Path, single_location: str, output_dir: Path) -> bool:
+    error = ''
+    if not template.exists():
+        error = f'\nError: the template file {template} does not exist.'
+    elif not locations_file.exists():
+        error = f'\nError: the locations file {locations_file} does not exist. Create locations.txt ' \
+                'or use the "-s" switch to specify a single location.'
+    elif not len(single_location) and not locations_file.stat().st_size:
+        error = f'\nError: the locations file {locations_file} is empty. Add locations ' \
+                'to this file, or, use the "-s" switch to specify a single location.'
+    elif not output_dir.exists():
+        error = f'\nError: the output director {output_dir} does not exist.'
+
+    if len(error):
+        print(error)
+
+    return True if 0 == len(error) else False
+
+
+@click.command()
+@click.option('-l', '--locations_file',
+                default=DEFAULT_LOCATIONS_FILE,
+                show_default=True,
+                type=click.Path(dir_okay=False),
+                help=f'The file with the location parameters for the template.')
+@click.option('-t', '--template',
+                default=DEFAULT_TEMPLATE_FILE,
+                show_default=True,
+                type=click.Path(dir_okay=False),
+                help='The model spec template file')
+@click.option('-s', '--single_location',
+                default='',
+                help='Specify a single location name. This takes precedence over the default locations file.')
+@click.option('-o', '--output_dir',
+                default=MODEL_SPEC_DIR,
+                show_default=True,
+                type=click.Path(dir_okay=True),
+                help='Specify an output directory. Directory must exist.')
+def make_specs(template: Path, locations_file : Path, single_location: str, output_dir: Path) -> None:
+    """
+    Generate model specifications based on a TEMPLATE. Supply the locations for which
+    you want a model spec generated by filling in the empty 'locations.txt' file.
+    The location file should contain a single location per line and locations must
+    be formatted so that they match location names in GBD location set X
+    """
+    template = Path(template)
```

Comment:
I recommend you expanduser and resolve your paths when they come from the cli.  It makes the tools much more flexible.